app/src/main/java/com/example/studentarena/fragments/FeedFragment.java

```diff
@@ -69,16 +71,27 @@ public void onClick(View v) {
         // set the adapter on the recycler view
         rvPosts.setAdapter(adapter);
         queryPosts();
+        swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
+        // Setup refresh listener which triggers new data loading
+        swipeContainer.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                queryPosts();
+            }
+        });
+        // Configure the refreshing colors
+        swipeContainer.setColorSchemeResources(android.R.color.holo_blue_bright,
+                android.R.color.holo_green_light,
+                android.R.color.holo_orange_light,
+                android.R.color.holo_red_light);
     }
 
     private void queryPosts() {
+        allPosts.clear();
         // specify what type of data we want to query - Post.class
         ParseQuery<Post> query = ParseQuery.getQuery(Post.class);
         // include data referred by user key
-        query.include(Post.KEY_USER);
-        query.setLimit(20);
-        // order posts by creation date (newest first)
-        query.addDescendingOrder("createdAt");
+        query.include(Post.KEY_USER).setLimit(20).addDescendingOrder("createdAt");
         // start an asynchronous call for posts
         query.findInBackground(new FindCallback<Post>() {
```

Comment:
you can streamline this even further:
```
ParseQuery.getQuery(Post.class)
    .include(Post.KEY_USER)
    .setLimit(20)
    .addDescendingOrder("createdAt")
    .findInBackground(new FindCallback<Post>() {
        ...
    });
```

if you think about what your `query` variable is, it doesn't really contain any data. it's just the way Parse wants us to tell it what query it should run under the hood and where it should put the results later on when it's done. we can accomplish that without ever having to store that `ParseQuery<Post>` in a variable or give it a name

not sure if this will be helpful or not but consider this fake example code that calculates the number of seconds in a day:
```
// Extra verbose
sixty = 60;
secondsInDay = sixty.times(60).times(60).times(24);

// Streamlined
secondsInDay = 60.times(60).times(60).times(24);
```